### DISCUSSION TITLE
How to preventing Memory Leaks on multiple clickhouse-js requests

### DISCUSSION THREAD

**Discussion by calebeaires** [ClickHouse/clickhouse-js#170] [Q&A]
During my experimentation with this library, I discovered a memory leak issue when a web application makes multiple requests while keeping the same connection setup. 

To investigate further, I conducted a test using a simple script to assess the ability of clickhouse-js to handle thousands of subsequent requests. Surprisingly, all requests were successful and responded without any issues.
However, over time, I noticed that the memory usage of the application continued to increase and did not stabilize.

To address this problem, **I would like to know how to prevent memory leaks when utilizing clickhouse-js in my web application**.

**Comment by slvrtrn** [comment]
Sorry for the late response; TIL discussions do not send me any notifications.
Can you please share the code that illustrates the memory leak?
